core/domain/exp_services.py

```diff
@@ -1912,11 +1912,15 @@ def regenerate_missing_stats_for_exploration(exp_id):
                     prev_exp.state_interaction_ids_dict[state_name])
                 current_interaction_id = (
                     exp.state_interaction_ids_dict[state_name])
-                exp_stats_list[i].state_stats_mapping[state_name] = (
-                    prev_exp_stats.state_stats_mapping[
-                        prev_state_name].clone()
-                    if current_interaction_id == prev_interaction_id else
-                    stats_domain.StateStats.create_default())
+                if current_interaction_id != prev_interaction_id or (
```

Comment:
This is a bit hard to understand, can you explain why we need to create a state here (maybe with a comment)?

Reply:
Done!